sdk/js/src/service/devices/index.js

```diff
@@ -120,13 +120,21 @@ class Devices {
       ? requestTreeOverwrite
       : splitSetPaths(paths, mergeBase)
 
-    // Retrieve join information if not present
-    if (!create) {
-      const paths = [['application_server_address'], ['network_server_address']]
+    if (create) {
+      // Skip request to JS if an external js is used
+      if (!Boolean(device.join_server_address)) {
```

Comment:
Is this testing whether `join_server_address` is empty, to indicate if it's external? External can be empty address or other (not cluster) address.